pennylane/tape/qnode.py

```diff
@@ -815,13 +847,32 @@ def circuit(weights):
     """
 
     def _metric_tensor_fn(*args, **kwargs):
+        try:
+            jac = qml.math.stack(_get_classical_jacobian(qnode)(*args, **kwargs))
+            jac = qml.math.reshape(jac, [qnode.qtape.num_params, -1])
+            perm = np.argsort(np.argsort(np.nonzero(qml.math.toarray(jac))[1]))

```

Comment:
Does it make sense to pass `qml.math` values to raw `np` methods? It will cast them to numpy instead of the underlying tensor type, and won't work in `jit` contexts.

Reply:
I suppose not, was trying to avoid creating new `qml.math.argsort` and `qml.math.nonzero` functions 😬